src/test/regress/expected/propagate_extension_commands.out

```diff
@@ -0,0 +1,285 @@
+RESET client_min_messages;
+-- now CREATE EXTENSION within a transction block and with version defined and commit it
+BEGIN;
+	CREATE EXTENSION isn WITH VERSION '1.1' SCHEMA public;
+	-- now, create a reference table relying on the data types
+	CREATE TABLE ref_table (a public.issn);
+	SELECT create_reference_table('ref_table');
+ create_reference_table 
+------------------------
+ 
+(1 row)
+
+	-- now, update to a newer version
+	ALTER EXTENSION isn UPDATE TO '1.2';
+	-- now change the schema
+	ALTER EXTENSION isn SET SCHEMA "extension'test";
+COMMIT;
+-- make sure that the object is distributed 
+SELECT count(*) FROM citus.pg_dist_object WHERE objid = (SELECT oid FROM pg_extension WHERE extname = 'isn');
+ count 
+-------
+     1
+(1 row)
+
+-- show that the CREATE and ALTER EXTENSION commands are propagated as we committed the changes
+SELECT run_command_on_workers($$SELECT extnamespace FROM pg_extension WHERE extname = 'isn'$$);
+  run_command_on_workers  
+--------------------------
+ (localhost,57637,t,2200)
+ (localhost,57638,t,2200)
+(2 rows)
+
+SELECT run_command_on_workers($$SELECT extversion FROM pg_extension WHERE extname = 'isn'$$);
+ run_command_on_workers  
+-------------------------
+ (localhost,57637,t,1.1)
+ (localhost,57638,t,1.1)
+(2 rows)
+
+-- SET client_min_messages TO WARNING before executing a DROP EXTENSION statement
+SET client_min_messages TO WARNING;
+-- drop the extension finally
+DROP EXTENSION isn CASCADE;
+-- restore client_min_messages after DROP EXTENSION
+RESET client_min_messages;
+drop table ref_table;
+drop table test_table;
+-- now make sure that the reference tables depending on an extension can be succesfully created.
+-- we should also ensure that we replicate this reference table (and hence the extension) 
+-- to new nodes after calling master_add_node. 
+-- but as we have only 2 ports in postgresql tests, let's remove one of the nodes first
+SELECT master_remove_node('localhost', 57638);
+ master_remove_node 
+--------------------
+ 
+(1 row)
+
+select distinct logicalrelid from pg_dist_shard;
+ logicalrelid 
+--------------
+(0 rows)
+
+-- then create the extension
+CREATE EXTENSION hstore;
+-- show that the extension is created on existing worker
+SELECT run_command_on_workers($$SELECT extnamespace FROM pg_extension WHERE extname = 'hstore'$$);
+  run_command_on_workers   
+---------------------------
+ (localhost,57637,t,17692)
+(1 row)
+
+SELECT run_command_on_workers($$SELECT extversion FROM pg_extension WHERE extname = 'hstore'$$);
+ run_command_on_workers  
+-------------------------
+ (localhost,57637,t,1.6)
+(1 row)
+
+-- now create the reference table
+CREATE TABLE ref_table_2 (x hstore);
+SELECT create_reference_table('ref_table_2');
+ create_reference_table 
+------------------------
+ 
+(1 row)
+
+-- and add the other node
+SELECT master_add_node('localhost', 57638);
+NOTICE:  Replicating reference table "ref_table_2" to the node localhost:57638
```

Comment:
We need to make sure that this block works in a transaction block as well:
```SQL
BEGIN;
CREATE EXTENSION hstore;
CREATE TYPE test_type AS (a int, b hstore); 
CREATE TABLE t (a int, b test_type );
SELECT create_distributed_table('t', 'a');
SELECT master_add_node(...);
COMMIT;
```

which unfortunately fails, I'll look a bit more

Reply:
added this test